### DISCUSSION TITLE
Free security test report with AppSweep by GuardSquare

### DISCUSSION THREAD

**Discussion by OneNewSmartIdea** [horizontalsystems/unstoppable-wallet-android#5730] [General]
I ve generated a complete security report for the mobile wallet, hope its helpful,  you can inspect here:
https://appsweep.guardsquare.com/builds/f3f2e9f5-dc2f-4577-8c53-996f5602dd8e

I have also generated one for MetaMask (v5.8 because of the 256 Mb size limit)so you can compare it:
https://appsweep.guardsquare.com/builds/52ef5847-19c0-4c2e-b730-9fe6de6303b2

You can download also the PDF versions (they don t include the code). 
The one from MetaMask is 45 pages long and the one from Unstoppable is 147 pages.

https://appsweep.guardsquare.com/api/builds/52ef5847-19c0-4c2e-b730-9fe6de6303b2/reports/218a7e1f-9996-4b83-87d7-21a4d25ca727/download

https://appsweep.guardsquare.com/api/builds/f3f2e9f5-dc2f-4577-8c53-996f5602dd8e/reports/889ce55b-19d1-4789-a1ac-2066a9b3efe8/download

Please inform us if it helped.

**Comment by OneNewSmartIdea** [comment]
I ve also generated a report for Brave Browser (has a crypto wallet) v1.48.05 (newer versions were over 256 Mb limit)

https://appsweep.guardsquare.com/builds/ebb00e0a-f99e-4371-9ec4-1c53df46732a/owasp

82 pages PDF file:
https://appsweep.guardsquare.com/api/builds/ebb00e0a-f99e-4371-9ec4-1c53df46732a/reports/14afd5e8-9c04-4d34-b6e7-9f3ecefefa41/download

**Reply by abdrasulov** [reply]
Thanks for the report! We'll check the issues.

**Reply by OneNewSmartIdea** [reply]
Glad to help.

**Comment by OneNewSmartIdea** [comment]
I have done another security test report for the latest version and it can be seen clearly that improvements have been done ! 
https://appsweep.guardsquare.com/builds/4e7fc84d-2a34-43bc-ab93-f61111852f00?filter=%7B%22Severity%22%3A%22high%22%7D&page=1
In order to be able to use the comparison function to see the differences make an account there, create a project and add the last 2 versions of the wallet app to the project.
I m curious if the changes done were made after the devs read my previous report or it just happened they addresses those issues ?

**Reply by rafaelekol** [reply]
We started working on this report yesterday. I am not sure what changes made the difference in reports.

**Comment by Cobra2911** [comment]
ccol
